### DISCUSSION TITLE
Upgrade builder bison & gcc

### DISCUSSION THREAD

**Discussion by trexxet** [community/community#25925] [Actions]
<p>My project is using feature introduced in Bison 3.6 (<code>%define parse.error custom</code>). However, <code>ubuntu-latest</code> environment has the 3.0 as newest and <code>ubuntu-20.04</code> has 3.5.</p>
<p>Also, with the introduction of C++20, it would be nice to use GCC 10.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/trexxet">@trexxet</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>You can update the version to bison3.6 and install GCC10 on the hosted runner, code sample as below:</p>
<pre><code class="lang-auto">      - name: update bison
        run: |
          wget http://ftp.gnu.org/gnu/bison/bison-3.6.tar.gz
          tar -zxvf bison-3.6.tar.gz
          cd bison-3.6
          ./configure
          make
          sudo make install         
      - name: update GCC
        run: |
          sudo apt-get install build-essential software-properties-common 
          sudo add-apt-repository ppa:ubuntu-toolchain-r/test
          sudo apt-get install gcc-10 g++-10
          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 100 --slave /usr/bin/g++ g++ /usr/bin/g++-10
          sudo update-alternatives --config gcc
      - name: check version
        run: |
          bison -V
          gcc --version
</code></pre>
<p>Please check my workflow for your reference.<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/weide-zhou/ticket13/runs/913391284?check_suite_focus=true#step:4:14" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181094147-adde5fdb-65e5-432c-b643-d3810debdf7a.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/weide-zhou/ticket13/runs/913391284?check_suite_focus=true#step:4:14" target="_blank" rel="nofollow noopener">weide-zhou/ticket13</a></h3>

<p>Contribute to weide-zhou/ticket13 development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The hosted runner image is updated weekly, if you’d like the updated version to be default, you can raise a feedback ticket in below link, github product manager will take a review and confirm.<br>
<a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" class="onebox" target="_blank" rel="nofollow noopener">https://support.github.com/contact/feedback?contact[category]=actions</a></p>
<p>Thanks</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/trexxet">@trexxet</a>,</p>
<p>Just want to know if the issue is resolved with my answer? Let me know if you have any other questions.</p>
<p>If it’s resolved, you can mark the answer to archive the ticket.</p>
<p>Thanks.</p>

**Comment by trexxet** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>,<br>
yep, that works. Thanks!</p>
